ipa-core/src/test_fixture/ipa.rs

```diff
@@ -253,7 +259,36 @@ pub async fn test_oprf_ipa<F>(
 
     //TODO(richaj): To be removed once the function supports non power of 2 breakdowns
     let _ = result.split_off(expected_results.len());
-    assert_eq!(result, expected_results);
+
+    match dp_params {
+        DpParams::NoDp => {
+            assert_eq!(result, expected_results);
+        }
+        DpParams::WithDp { epsilon } => {
+            let per_user_credit_cap_f64 = f64::from(config.per_user_credit_cap);
+            let noise_params = NoiseParams {
+                epsilon,
+                delta: 1e-6,
+                ell_1_sensitivity: per_user_credit_cap_f64,
+                ell_2_sensitivity: per_user_credit_cap_f64,
+                ell_infty_sensitivity: per_user_credit_cap_f64,
+                ..Default::default()
+            };
+
+            let num_bernoulli = crate::protocol::dp::find_smallest_num_bernoulli(&noise_params);
+            let mean: f64 = f64::from(num_bernoulli) * 0.5; // n * p
+            let standard_deviation: f64 = (f64::from(num_bernoulli) * 0.5 * 0.5).sqrt(); //  sqrt(n * (p) * (1-p))
+            for (index, sample) in result.iter().enumerate() {
+                assert!(
+                    f64::from(*sample) - mean
+                        > f64::from(expected_results[index]) - 5.0 * standard_deviation
+                        && f64::from(*sample) - mean
+                        < f64::from(expected_results[index]) + 5.0 * standard_deviation
```

Comment:
This is not obviously correct, because it is a little tricky to read.

I think that you could have a simpler test here:

```rust
assert_eq!(result.len(), expected_results.len());
for (index, (&sample, &expected)) in zip(result.iter(), expected_results.iter()).enumerate() {
  assert!(abs(f64::from(sample) - mean - f64::from(expected)) < 5.0 * stddev,
              "DP result was not within 5 standard deviations from what was expected");
}
```

The next question is whether 5 is the right number.  If we run this 1000 times a day for 3 years, it looks like we'll get one value outside of the range in that time, which seems workable.

Reply:
Thanks, that's cleaner.